Core/automation/lib/python/core/utils.py

```diff
@@ -118,6 +122,61 @@ def validate_uid(uid):
     uid = re.sub(r"__+", "_", uid)
     return uid
 
+def get_item_value(item_or_item_name, return_type=None, default=None, unit=None):
+    """
+    Returns the Item's value if the Item exists and is initialized, otherwise default. 
+    Returned item type will be dependent on the item's type. If return_type is supplied,
+    the Item will be returned as that type if possible, otherwise the default will be 
+    returned. Item value will be converted to the provided unit type if possible; otherwise,
+    the default will be returned. If no default is supplied and an Item value cannot be 
+    returned, None is returned 
+
+    Args:
+        item_or_item_name (Item or str): name of the Item
+        return_type (int, float, str): item type to be returned. If the Item cannot be 
+            converted to return_type, default will be returned 
+        default (int, float, ON, OFF, OPEN, CLOSED, str, DateTime): the default
+            value to be returned if the Item is non-existent/uninitialized or cannot
+            be converted to return_type
+        unit (QuantityType): Convert value of Item to specified units
+            https://www.openhab.org/docs/concepts/units-of-measurement.html
+
+    Returns:
+        int, float, ON, OFF, OPEN, CLOSED, str, DateTime, or None: the state of the Item
+            as a suitable type, or converted to return_type if given; or the default value 
+            if the Item's state cannot be converted or is NULL or UNDEF; otherwise None
+    """
+    
+    type_methods = {int: "intValue", float: "floatValue", str: "toString"}
+    item = validate_item(item_or_item_name) # type: t.Union[GenericItem, None]
+    if item:
+        state = item.getState() # type: State
+        if isinstance(state, QuantityType) and unit is not None:
+            try:
+                state = state.toUnit(unit)
+            except:
+                LOG.warn(u"Item '{}' cannot be converted to '{}' units, returning default value".format(item.name, unit))
```

Comment:
Good idea!

Let's add *state* to the message and shorten it to just describe what went wrong, returning the default is implied by the docstring already.
```python
LOG.warn(u"Item '{}' state [{}] cannot be converted to '{}' units".format(item.name, item.state, unit))
```

Reply:
Cool, I'm shortening the failed conversion warning similarly.

```python
LOG.warn(u"Item '{}' type '{}', cannot be converted to '{}'".format(item.name, type(state), return_type)
```